contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/graphics_test.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import subprocess
+import argparse
+import re
+import time
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def debug_logging(verbose=False):
+    """Configure logging output.
+
+    Args:
+        verbose: If True, set logging level to DEBUG; otherwise INFO
+    """
+    level = logging.DEBUG if verbose else logging.INFO
+    logging.basicConfig(
+        level=level, format="%(asctime)s - %(levelname)s - %(message)s"
+    )
+
+
+def is_ubuntu_frame_active():
+    """Check if ubuntu-frame is active."""
+    try:
+        subprocess.check_output(["pgrep", "-if", "ubuntu-frame"])
+        logger.info("The ubuntu-frame is active")
+        return True
+    except subprocess.CalledProcessError:
+        logger.info("ubuntu-frame is not active")
+        return False
+
+
+def test_ubuntu_frame_launching():
+    """Test ubuntu-frame launching."""
+    if is_ubuntu_frame_active():
+        logger.info("No need to bring it up again")
+        logger.info("journal log of ubuntu frame:")
+        subprocess.run(["journalctl", "-b", "0", "-g", "ubuntu-frame"])
+    else:
+        logger.info("Activating ubuntu-frame now...")
+        try:
+            subprocess.run(
+                ["timeout", "20s", "ubuntu-frame"],
+                check=True,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+        except subprocess.CalledProcessError as e:
+            if e.returncode == 124:
+                logger.info(
+                    "\nPASS: Timeout reached without any failures detected."
+                )
+            else:
+                return 1
+    return 0
+
+
+def test_glmark2_es2_wayland():
+    """Test glmark2-es2-wayland."""
+    cmd = [
+        "env",
+        "XDG_RUNTIME_DIR=/run/user/0",
+        "graphics-test-tools.glmark2-es2-wayland",
+    ]
+    exit_code = 0
+
+    frame_pid = None
+    if not is_ubuntu_frame_active():
+        logger.info("Activating ubuntu-frame now...")
+        proc = subprocess.Popen(
+            ["ubuntu-frame"],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+        frame_pid = proc.pid
+        time.sleep(10)
+
+    logger.info("Running glmark2-es2-wayland benchmark...")
+    process = subprocess.Popen(
+        cmd,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        universal_newlines=True,
+    )
+    output = ""
+    for line in process.stdout:
+        logger.info(line.strip())
+        output += line
+
+    process.wait()
```

Comment:
I think you can directly call `subprocess.check_output` here since we block the main thread until glmark2 is done